include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -125,12 +123,228 @@ class NBL_API IShaderCompiler : public core::IReferenceCounted
 
 		virtual IShader::E_CONTENT_TYPE getCodeContentType() const = 0;
 
-		IIncludeHandler* getIncludeHandler() { return m_inclHandler.get(); }
-		const IIncludeHandler* getIncludeHandler() const { return m_inclHandler.get(); }
+		class NBL_API IIncludeLoader : public core::IReferenceCounted
+		{
+		public:
+			virtual std::string getInclude(const system::path& searchPath, const std::string& includeName) const = 0;
+		};
+
+		class NBL_API IIncludeGenerator : public core::IReferenceCounted
```

Comment:
btw you can delete any `NBL_API` in the files you're retouching, only `NBL_API2` actually does anything for DLLs